// @intelica/odoo-xmlrpc/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ERRORS = exports.Categories = exports.Employees = exports.Leads = exports.Contacts = exports.MODEL_TYPE = exports.OdooCRUD = exports.Odoo = void 0;
var odoo_1 = require("./controllers/odoo");
Object.defineProperty(exports, "Odoo", { enumerable: true, get: function () { return odoo_1.Odoo; } });
var odoo_crud_1 = require("./controllers/odoo.crud");
Object.defineProperty(exports, "OdooCRUD", { enumerable: true, get: function () { return odoo_crud_1.OdooCRUD; } });
var odoo_types_1 = require("./models/odoo.types");
Object.defineProperty(exports, "MODEL_TYPE", { enumerable: true, get: function () { return odoo_types_1.MODEL_TYPE; } });
var contacts_odoo_1 = require("./controllers/contacts.odoo");
Object.defineProperty(exports, "Contacts", { enumerable: true, get: function () { return contacts_odoo_1.Contacts; } });
var leads_odoo_1 = require("./controllers/leads.odoo");
Object.defineProperty(exports, "Leads", { enumerable: true, get: function () { return leads_odoo_1.Leads; } });
var employees_odoo_1 = require("./controllers/employees.odoo");
Object.defineProperty(exports, "Employees", { enumerable: true, get: function () { return employees_odoo_1.Employees; } });
var categories_odoo_1 = require("./controllers/categories.odoo");
Object.defineProperty(exports, "Categories", { enumerable: true, get: function () { return categories_odoo_1.Categories; } });
var errors_enum_1 = require("./constants/errors.enum");
Object.defineProperty(exports, "ERRORS", { enumerable: true, get: function () { return errors_enum_1.ERRORS; } });
